### PR TITLE
Update changelog more consistently

### PR DESCRIPTION
**Pull request by agjohnson**
* Get the last modified date from git, not local file as normally expected
* Make sure we're on master to run
* Don't do reset to master on release

### CODE REVIEW

**Comment by agjohnson** [comment]
More unrelated errors

File: tasks.py [hunk 1]
```diff
@@ -33,6 +41,11 @@ def prepare(ctx, version):
         config.write(configfile)
 
     print('Installing github-changelog')
+    # Get last modified date from Git instead of assuming the file metadata is
+    # correct. This can change depending on git reset, etc.
+    git_log = ctx.run('git log -1 --format="%ad" -- CHANGELOG.rst')
+    last_modified = parse(git_log.stdout.strip()).strftime('%Y-%m-%d')
```
Comment (ericholscher) [inline tasks.py]: We should put a note in the changelog that it's auto-generated, and not to touch it manually.

Comment (agjohnson) [reply tasks.py]: Actually, one of the benefits of this tool is that it only does that last release. It does allow for authoring in between generating and commiting, so we can add deployment and migration notes, or more general description of some of the changes

File: tasks.py [hunk 2]
```diff
@@ -24,6 +25,13 @@ def prepare(ctx, version):
     updated.  New entries will end up at the top of the file, under a heading
     for the new version.
     """
+    # Ensure we're on the master branch first
+    git_rev_parse = ctx.run('git rev-parse --abbrev-ref HEAD', hide=True)
+    current_branch = git_rev_parse.stdout.strip()
+    if current_branch != 'rel':
```
Comment (humitos) [inline tasks.py]: why `rel` here? shouldn't be `master`?

I mean, it checks if we are in `rel` and if not, it says "hey, you need to be on master"

Other, we want to ensure we are on `rel` not on `master`, right?

**Pull request merged**